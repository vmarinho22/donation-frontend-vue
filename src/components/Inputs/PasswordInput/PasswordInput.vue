<template>
  <v-text-field
    :append-inner-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
    :rules="[
      (v) => {
        const parsedValue = z
          .string({
            required_error: props.labels.rules.required,
          })
          .min(8, {
            message: props.labels.rules.min,
          })
          .safeParse(v)

        if (!parsedValue.success) {
          return parsedValue.error.errors.map((item) => item.message).join('|')
        }

        return parsedValue.success
      }
    ]"
    :type="showPassword ? 'text' : 'password'"
    :label="props.labels.label"
    :hint="props.labels.hint"
    counter
    @click:append-inner="showPassword = !showPassword"
    v-bind="attrs"
  />
</template>

<script setup lang="ts">
import { useAttrs } from 'vue'
import  { ref } from 'vue'
import { z } from 'zod'

const props = defineProps<{
  labels: {
    label: string
    hint: string
    rules: {
      required: string
      min: string
    }
  }
}>()

const attrs = useAttrs()

const showPassword = ref(false)
</script>