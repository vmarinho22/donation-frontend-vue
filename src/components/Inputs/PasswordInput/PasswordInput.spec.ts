import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import PasswordInput from './PasswordInput.vue'
import vuetify from '@/tests/plugins/vuetify'

describe('PasswordInput', () => {
  const labelsProp = {
    label: 'Test',
    hint: 'Test',
    rules: {
      required: 'Test',
      min: 'Test'
    }
  }
  it('renders the component', () => {
    const wrapper = mount(PasswordInput, {
      global: {
        plugins: [vuetify]
      },
      props: {
        labels: labelsProp
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('show password', async () => {
    const wrapper = mount(PasswordInput, {
      global: {
        plugins: [vuetify]
      },
      props: {
        labels: labelsProp
      }
    })

    const button = wrapper.find('.mdi-eye-off')

    await button.trigger('click')

    const input = wrapper.find('input')

    expect(wrapper.html()).toMatchSnapshot()
    expect(input.attributes('type')).toBe('text')
  })
})
