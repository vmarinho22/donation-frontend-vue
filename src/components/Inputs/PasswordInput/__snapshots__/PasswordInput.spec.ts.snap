// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordInput > renders the component 1`] = `
"<div class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field\\">
  <!---->
  <div class=\\"v-input__control\\">
    <div class=\\"v-field v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr\\">
      <div class=\\"v-field__overlay\\"></div>
      <div class=\\"v-field__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class=\\"v-field__field\\" data-no-activator=\\"\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-0\\">
          <!---->Test
        </label><label class=\\"v-label v-field-label\\" for=\\"input-0\\">
          <!---->Test
        </label>
        <!----><input size=\\"1\\" type=\\"password\\" id=\\"input-0\\" aria-describedby=\\"input-0-messages\\" class=\\"v-field__input\\">
        <!---->
      </div>
      <!---->
      <div class=\\"v-field__append-inner\\">
        <!----><i class=\\"mdi-eye-off mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Test appended action\\"></i>
      </div>
      <div class=\\"v-field__outline\\">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class=\\"v-input__details\\">
    <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-0-messages\\">
      <!---->
    </transition-group-stub>
    <!----><span></span>
    <transition-stub name=\\"slide-y-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
      <div class=\\"v-counter\\" style=\\"display: none;\\">0</div>
    </transition-stub>
  </div>
</div>"
`;

exports[`PasswordInput > show password 1`] = `
"<div class=\\"v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field\\">
  <!---->
  <div class=\\"v-input__control\\">
    <div class=\\"v-field v-field--active v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-field--focused v-locale--is-ltr\\">
      <div class=\\"v-field__overlay\\"></div>
      <div class=\\"v-field__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class=\\"v-field__field\\" data-no-activator=\\"\\"><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-0\\">
          <!---->Test
        </label><label class=\\"v-label v-field-label\\" for=\\"input-0\\">
          <!---->Test
        </label>
        <!----><input size=\\"1\\" type=\\"text\\" id=\\"input-0\\" aria-describedby=\\"input-0-messages\\" class=\\"v-field__input\\">
        <!---->
      </div>
      <!---->
      <div class=\\"v-field__append-inner\\">
        <!----><i class=\\"mdi-eye mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Test appended action\\"></i>
      </div>
      <div class=\\"v-field__outline\\">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class=\\"v-input__details\\">
    <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-0-messages\\">
      <div class=\\"v-messages__message\\">Test</div>
    </transition-group-stub>
    <!----><span></span>
    <transition-stub name=\\"slide-y-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
      <div class=\\"v-counter\\" style=\\"\\">0</div>
    </transition-stub>
  </div>
</div>"
`;
